### PR TITLE
Allow to specify psycopg* in extras and switch to `build`

### PR DESCRIPTION
**Pull request by CyberDem0n**
* remove check_psycopg() call from the setup.py, when installing from wheel it doesn't work anyway.
* call check_psycopg() function before process_arguments(), because the last one is trying to import psycopg and fails with the stacktrace, while the first one shows a nice human-readable error message.
* add psycopg2, psycopg2-binary, and psycopg3 extras, that will install psycopg2>=2.5.4, psycopg2-binary, or psycopg[binary]>=3.0.0 modules respectively.
* move check_psycopg() function to the __main__.py.
* introduce the new extra called `all`, it will allow to install all dependencies at once (except psycopg related).
* use the `build` module in order to create sdist bdist_wheel packages.
* update the documentation regarding psycopg and extras (dependencies).

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -73,12 +60,20 @@ raft
     `pysyncobj` module in order to use python Raft implementation as DCS
 aws
     `boto3` in order to use AWS callbacks
+all
+    all of the above (except psycopg family)
+psycopg
+    `psycopg[binary]>=3.0.0` module
```
Comment (barthisrael) [inline docs/installation.rst:66]: This line differs from the corresponding one in `README.rst`.
I guess we should adjust either of them, probably in `README.rst` given the commit message.

File: setup.py [hunk 2]
```diff
@@ -139,15 +145,22 @@ def setup_package(version):
                     deps[i] = r
                     EXTRAS_REQUIRE[e] = deps
                     extra = True
-                    break
-            if extra:
-                break
         if not extra:
             install_requires.append(r)
 
+    # Just for convinience, if someone wants to install dependencies for all extras
```
Comment (barthisrael) [inline setup.py:151]:
```suggestion
    # Just for convenience, if someone wants to install dependencies for all extras
```

**Review by barthisrael — APPROVED** [review]
Looks good!
I only have a couple small pointers.

File: docs/installation.rst (hunk 1 above)
Comment (barthisrael) [reply docs/installation.rst:66]: The code in `setup.py` and the psycopg3 docs page also suggest the one to be fixed is `README.rst`.

**Comment by barthisrael** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**